appservice/src/createKuduClient.ts

```diff
@@ -27,7 +27,7 @@ export async function createKuduClient(context: IInternalKuduActionContext, site
 
         const clientOptions = { baseUri: site.kuduUrl, userAgent: appendExtensionUserAgent };
 
-        let credentials = site.subscription.credentials;
+        let serviceClientCredentials: ServiceClientCredentials = site.subscription.credentials;
```

Comment:
Typing thing due to the change. Both `AzExtServiceClientCredentials` and `BasicAuthenticationCredentials` extend `ServiceClientCredentials`, but `BasicAuthenticationCredentials` does _not_ extend `AzExtServiceClientCredentialsT2`. However, Kudu doesn't need that. The fix is just to tell TypeScript we only care about the `ServiceClientCredentials` half of the interface.